File: modules/Plugin/src/daybreak/abilitywar/utils/base/minecraft/server/ServerType.java
package daybreak.abilitywar.utils.base.minecraft.server;

import com.google.common.base.Enums;
import org.bukkit.Bukkit;

public enum ServerType {

	UNKNOWN,
	CRAFTBUKKIT,
	SPIGOT,
	PAPER;

	private static class Companion {
		public static final ServerType INSTANCE;

		static {
			final String[] split = Bukkit.getVersion().split("-");
			final String name = split[1].toUpperCase();
			INSTANCE = Enums.getIfPresent(ServerType.class, name).or(name.equals("BUKKIT") ? CRAFTBUKKIT : UNKNOWN);
		}
	}

	public static ServerType getServerType() {
		return Companion.INSTANCE;
	}

	public static boolean is(final ServerType other) {
		return Companion.INSTANCE == other;
	}

}

File: modules/Plugin/src/daybreak/abilitywar/utils/base/minecraft/server/ServerType.kt
package daybreak.abilitywar.utils.base.minecraft.server

import com.google.common.base.Enums
import org.bukkit.Bukkit

enum class ServerType {

	UNKNOWN,
	CRAFTBUKKIT,
	SPIGOT,
	PAPER;

	companion object {

		private val INSTANCE: ServerType = with(Bukkit.getVersion().split("-")) {
			if (this.size >= 2) {
				val name = this[1].toUpperCase()
				Enums.getIfPresent(ServerType::class.java, name).or(if (name == "BUKKIT") CRAFTBUKKIT else UNKNOWN)
			} else UNKNOWN
		}

		@JvmStatic
		fun getServerType(): ServerType {
			return INSTANCE
		}

		@JvmStatic
		fun `is`(other: ServerType): Boolean {
			return INSTANCE == other
		}

	}

}